app/controllers/app.py

```diff
@@ -842,6 +814,110 @@ def patch(self, app_id):
                          a_app_id=app_id)
             return dict(status='fail', message=str(exc)), 500
 
+class AppRevisionsView(Resource):
+    @jwt_required
+    def get(self, app_id):
+        """
+        get application revisions
+        """
+        try:
+            current_user_id = get_jwt_identity()
+            current_user_roles = get_jwt_claims()['roles']
+
+            # get pagination params
+            page = request.args.get('page', 1, type=int)
+            per_page = request.args.get('per_page', 10, type=int)
+
+            app_schema = AppSchema()
+
+            app = App.get_by_id(app_id)
+
+            if not app:
+                return dict(status='fail', message=f'App {app_id} not found'), 404
+
+            project = app.project
+
+            if not project:
+                return dict(status='fail', message='Internal server error'), 500
+
+            if not is_owner_or_admin(project, current_user_id, current_user_roles):
+                if not is_authorised_project_user(project, current_user_id, 'member'):
+                    return dict(status='fail', message='Unauthorised'), 403
+
+            app_data, errors = app_schema.dumps(app)
+
+            app_list = json.loads(app_data)
+
+            cluster = Cluster.get_by_id(project.cluster_id)
+
+            if not cluster:
+                return dict(
+                    status='fail',
+                    message=f'cluster with id {project.cluster_id} does not exist'), 404
+
+            kube_host = cluster.host
+            kube_token = cluster.token
+            kube_client = create_kube_clients(kube_host, kube_token)
+
+            app_status_object = \
+                kube_client.appsv1_api.read_namespaced_deployment_status(
+                    app_list['alias'] + "-deployment", project.alias)
+
+            app_list["image"] = app_status_object.spec.template.spec.containers[0].image
+            app_list["port"] = app_status_object.spec.template.spec.containers[0].ports[0].container_port
+            app_list["replicas"] = app_status_object.spec.replicas
+            app_list["revision"] = app_status_object.metadata.annotations.get(
+                'deployment.kubernetes.io/revision')
```

Comment:
I don't think you are required to check `lines 865 to 870`, that's not part of the revision information this endpoint should return

Reply:
Thanks for highlighting that, but he line 869-870 is needed in subsequent processing to determine the current revision of the application i.e. 
`if app_list["revision"] == item.metadata.annotations.get('deployment.kubernetes.io/revision'):
                    replica_set["current"] = True
                    app_list["revision_id"] = int(
                        item.metadata.creation_timestamp.timestamp())`